annotation/functions.scm

```diff
@@ -121,13 +121,114 @@
    ; res is list of the GO terms directly related to 
    ; the input gene (g) that are members of the input namespaces
    (define res (find-memberln g namespaces))
-
+   (define go-regulates (append-map (lambda (go) (find-go-plus go regulates part-of)) res))
    (define all-parents (loop num-parents res '()))
 
-   (cons (node-info g) all-parents)
+   (append (node-info g) all-parents go-regulates)
+)
+
+(define regulates-rln (List (Concept "GO_regulates") (Concept "GO_positively_regulates") (Concept "GO_negatively_regulates")))
+
+(define (do-find-regulates input-atom)
+      (append (append-map (lambda (reg) (run-query (Bind 
+            (Evaluation
+               (Predicate (cog-name reg))
+               (ListLink
+                  (gar input-atom)
+                  (Variable "$go")
+               )
+            )
+
+            (Evaluation
+               (Predicate (cog-name reg))
+               (ListLink
+                  (gar input-atom)
+                  (Variable "$go")
+               )
+            )
+
+         ))) (cog-outgoing-set (gdr input-atom)))
+
+         (append-map (lambda (reg) (run-query (Bind 
+            (Evaluation
+               (Predicate (cog-name reg))
+               (ListLink
+                  (Variable "$go")
+                  (gar input-atom)
+               )
+            )
+
+            (Evaluation
+               (Predicate (cog-name reg))
+               (ListLink
+                  (Variable "$go")
+                  (gar input-atom)
+               )
+            )
+
+         ))) (cog-outgoing-set (gdr input-atom))))
+   )
+
+(define find-go-regulates (memoize-function-call do-find-regulates))
+
+(define (do-find-part-of go-term)
+   (append
+      (run-query (Bind
+         (Evaluation
+            (Predicate "has_part")
+            (List
+               go-term
+               (Variable "$go")
+            )
+         )
+         (Evaluation
+            (Predicate "has_part")
+            (List
+               go-term
+               (Variable "$go")
+            )
+         )
+      ))
+      (run-query
+         (Bind
+            (Evaluation
+               (Predicate "has_part")
+               (List
+                  (Variable "$go")
+                  go-term
+               )
+            )
+            (Evaluation
+               (Predicate "has_part")
+               (List
+                  (Variable "$go")
+                  go-term
+               )
+            )
```

Comment:
Same here https://github.com/MOZI-AI/annotation-scheme/pull/194/files#diff-aed3d7e3a5790c8e073ca5611e02a67fR192-R207, reverse search is not required